components/Weather.js

```diff
@@ -33,72 +54,75 @@ export default class Weather extends React.Component {
           // See error code charts below.
           console.log(error.code, error.message);
         },
-        {enableHighAccuracy: true, timeout: 15000},
+        { enableHighAccuracy: true, timeout: 15000 },
       );
     }
   };
-  _parseWeatherData = (data) => {
+
+  parseWeatherData = (data) => {
     const weather = {};
-    weather.city = datas.city.name;
+    weather.city = data.city.name;
     weather.dataPerHour = [];
-    for (const data of datas.list) {
+    data.forEach((weatherLog) => {
```

Comment:
The same thing as previous one.

Reply:
You didn't change, use map function here instead of forEach. https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/map

Reply:
I was convinced it was an object. I added a map method.